components/ReviewTransactionModal.tsx

```diff
@@ -12,43 +12,39 @@ import {
 import { Tooltip } from '@chakra-ui/tooltip';
 import { ethers } from 'ethers';
 import { parseUnits } from 'ethers/lib/utils.js';
+import { TokenListContextItem } from '@/contexts/TokenContext';
 import useNotifications from '@/hooks/useNotifications';
 import useRailgunTx from '@/hooks/useRailgunTx';
 import { shortenAddress } from '@/utils/address';
 
 type ReviewTransactionModalProps = {
   isOpen: boolean;
   onClose: () => void;
-  tokenAddress?: string;
   recipient: string;
-  tokenAmount?: string;
-  tokenDecimals?: number;
-  tokenSymbol: string;
-  tokenName: string;
+  token: TokenListContextItem;
+  amount: string;
+  onClick: () => void;
 };
 
 const ReviewTransactionModal = ({
   isOpen,
   onClose,
-  tokenAddress,
   recipient,
-  tokenAmount,
-  tokenDecimals,
-  tokenSymbol,
-  tokenName,
+  amount,
+  token,
+  onClick,
```

Comment:
what kind of click are we talking about here? might prefer a more descriptive name (onSubmitClick?)

Reply:
Good call will change